main.go

```diff
@@ -150,6 +154,21 @@ func main() {
 	ctx := ctrl.SetupSignalHandler()
 	setupIndexes(ctx, mgr, &cfg)
 
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(kubeConfig)
+	if err != nil {
+		setupLog.Error(err, "Unable to create the discovery client")
+		os.Exit(1)
+	}
+
+	serverVersionFetcher := kubeversion.NewServerVersionFetcher(discoveryClient, time.NewTicker(time.Second*10))
+
+	if err := serverVersionFetcher.FetchServerVersion(); err != nil {
+		setupLog.Error(err, "Unable to fetch server version")
+		os.Exit(1)
+	}
+
+	go serverVersionFetcher.FetchServerVersionPeriodic(ctx)
```

Comment:
Instead of creating a go routine here, you can let the manager run it for you.

You can make the ServerVersionFetcher implement `Start`, like we did for the scheduler https://github.com/kubernetes-sigs/kueue/pull/596

You would do `Start` there.

Reply:
+1

Reply:
Done